newspapers.com.js

```diff
@@ -35,46 +35,58 @@
 	***** END LICENSE BLOCK *****
 */
 
-
-// attr()/text() v2
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
-
-
-function detectWeb(doc, url) {
+function detectWeb(doc) {
 	return "newspaperArticle";
 }
 
-function doWeb(doc, url) {
+function doWeb(doc) {
 	var newItem = new Zotero.Item("newspaperArticle");
+	var scripts = doc.getElementsByTagName("script");
+	var json = '';
+	var jsonre = /var staPageDetail = JSON.parse\((.+?)\);/;
+	for (var i = 0; i < scripts.length; i++) {
+		var arr = scripts[i].textContent.match(jsonre);
+		if (arr) {
+			// we have to unescape the json string this way, since javascript
+			// unescape doesn't work here
+			json = arr[1];
+			break;
+		}
+	}
+	
+	// one JSON.parse to unstringify the json string, and one to parse it into an object
+	var details = JSON.parse(json);
+	if (typeof details === "string") {
+		details = JSON.parse(details);
+	}
+
 	var metaArr = {};
 	var metaTags = doc.getElementsByTagName("meta");
-	for (var i = 0 ; i < metaTags.length ; i++) {
+	for (i = 0; i < metaTags.length; i++) {
```

Comment:
This is still wrong — it would mean that if the `var i` above was changed or removed this would become a global assignment. These loops should all use either `for (let i = 0; …)` or `for (let foo of foos)`.

Reply:
I've gone with ```for (let metaTag in metaTags)```

Reply:
If you really mean `for … in`, that's incorrect — that does something different. See the two options I give above.